CMakeLists.txt

```diff
@@ -21,11 +21,14 @@ endif()
 message(STATUS "CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
 message(STATUS "CMAKE_SYSTEM_NAME: ${CMAKE_SYSTEM_NAME}")
 
+if(NOT DEFINED EXTRA_FLAGS)
+  set(EXTRA_FLAGS "-march=haswell" CACHE STRING
+          "Extra compilation flags for C and CXX." FORCE)
+endif()
+
 if (${CMAKE_BUILD_TYPE} MATCHES Release)
-  set(EXTRA_FLAGS "-Ofast -march=native")
+  set(EXTRA_FLAGS "-Ofast ${EXTRA_FLAGS}")
```

Comment:
In `Release` mode, we should use `-march=native`. It is our way to say 'compile by using all possible optimizations'. I like the idea to be able to override that, but better to do that in other modes, like 'Generic` or `Debug`, but not `Release`.

Reply:
> we should use `-march=native`. It is our way to say 'compile by using all possible optimizations'

That's true but it will use all possible optimizations for the **current** machine.
I have no data how Seqwish users do it but in other projects I have worked on the build machine is not always the same as the machines where the software is deployed. For example the CI system is using `Skylake` CPU but we run the app on Alder lake. 
This is the reason I've proposed to make it possible the user to decide what `march` (and any other C/CXX flags) value to use.

But as I said in my earlier [comment](https://github.com/ekg/seqwish/pull/106#discussion_r1011502964) I also think that `native` should be used as default. E.g. in pggb you could set whichever march you need for any build type.

Reply:
Yes, I agree. I would like the default and `Release` to do the same thing, which is to optimize as much as possible for the current machine.

Reply:
I've changed the default to `-march=native`.